rqt_shell/src/rqt_shell/shell.py

```diff
@@ -85,9 +86,21 @@ def __init__(self, context):
         super(Shell, self).__init__(context)
         self._context = context
         self.setObjectName('Shell')
+        self._args = self._parse_args(context.argv())
 
         self._widget = None
 
+    def _parse_args(self, argv):
+        parser = argparse.ArgumentParser(prog='rqt_shell')
+        Shell.add_arguments(parser)
+        return parser.parse_args(argv)
+
+    @staticmethod
+    def add_arguments(parser):
+        group = parser.add_argument_group('Options for rqt_shell plugin')
+        group.add_argument('-i', '--init_script',
+                            help="load this script after startup in $SHELL")
```

Comment:
The help message should mention that the options doesn't work with the "SimpleShell" backend.

Since I don't have Qt 4 available I couldn't try either of the two working backends. But reading the patch it looks like SpiderShell is sourcing the passed script while XtermShell executes the script? Am I reading this wrong? Shouldn't they both do the same thing? The help text should probably also be more clear what happens with the script: sourcing or executing.